src/app/core/services/owner-meals.service.ts

```diff
@@ -0,0 +1,48 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { Meal } from '../interfaces/meal';
+import { AuthenticationService } from './authentication.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OwnerMealsService {
+  public mealSubject: BehaviorSubject<Meal[]> = new BehaviorSubject([]);
```

Comment:
This should be private, if it is public it can be accessed from other parts of the app right? 
ownerMeals could also be private because you are only using it inside of this service. Please make the same changes to the owner-restaurants.service.ts